### DISCUSSION TITLE
How can I monitor a private git docker repositry.

### DISCUSSION THREAD

**Discussion by usman557** [containrrr/watchtower#727] [Q&A] [answered]
Hi, 

We are publishing the docker images as packages to the docker github registry (docker.pkg.github.com). How can I use the watch tower to monitor the image on this registry?

Thanks

**Comment by simonaronsson** [comment, marked as answer]
Hi!

Just as you would any other registry. Just make sure you launch your containers using their fully qualified vanity-url, for instance: `docker.pkg.github.com/usman557/your-project/your-repo:latest`

Best,
Simme

**Reply by usman557** [reply]
Thanks for the reply @simskij but that is exactly what I am doing but the latest image is not being pulled down.  Below is the details. 

Command to run docker image:

`docker run docker.pkg.github.com/usman557/consoleapp1/consoleapp1:master`

Command to watch the container: 

`docker run -d --name watchtower -e REPO_USER=USERNAME -e REPO_PASS=PASSWORD -v /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower docker.pkg.github.com/usman557/consoleapp1/consoleapp1:master --debug --trace --interval 10
`

I am not getting the latest changes that I am making to my docker image. Can you please let me know what I am missing here?

**Reply by simonaronsson** [reply]
> Command to watch the container:
> ```
> docker run -d --name watchtower -e REPO_USER=USERNAME -e REPO_PASS=PASSWORD -v
> /var/run/docker.sock:/var/run/docker.sock containrrr/watchtower 
> docker.pkg.github.com/usman557/consoleapp1/consoleapp1:master --debug --trace --interval 10
> ```

If you want to do selective monitoring, then `docker.pkg.github.com/usman557/consoleapp1/consoleapp1:master` needs to be switched to the name of the container, rather than the full URL of the image. For instance:

```
 $ docker run --name console docker.pkg.github.com/usman557/consoleapp1/consoleapp1:master
 $ docker run -d \
      --name watchtower \
      -e REPO_USER=USERNAME \
      -e REPO_PASS=PASSWORD \
      -v /var/run/docker.sock:/var/run/docker.sock \
      containrrr/watchtower \
      console \
      --debug \
      --trace \
      --interval 10
```

If you'd rather have watchtower monitor everything in your docker instance, just remove that part altogether:

```
 $ docker run -d \
      --name watchtower \
      -e REPO_USER=USERNAME \
      -e REPO_PASS=PASSWORD \
      -v /var/run/docker.sock:/var/run/docker.sock \
      containrrr/watchtower \
      --debug \
      --trace \
      --interval 10
```

**Reply by samru-rai** [reply]
Do I really need `REPO_USER` and `REPO_PASS`? I don't want to provide my username and password in my docker-compose file. I want to use  `$HOME/.docker/config.json` that gets generated after login based on their [instructions](https://docs.github.com/en/packages/guides/configuring-docker-for-use-with-github-packages#authenticating-with-a-personal-access-token) using personal token.

**Reply by piksel** [reply]
@samru-rai yeah, that is the usual way of doing it. As long as you use the full image name for the docker image it should work with a mounted docker config.
